src/encoding.ts

```diff
@@ -230,7 +232,8 @@ export function extractTransformsFromEncoding(oldEncoding: Encoding<string>) {
       // now the field should refer to post-transformed field instead
       encoding[channel] = {
         field: vgField(channelDef),
-        type: channelDef.type
+        type: channelDef.type,
+        title: title(channelDef, config)
```

Comment:
Actually, if there is neither bin not timeUnit, we shouldn't have to add this custom title.